dashboard/src/components/widget/footer/che-footer.html

```diff
@@ -16,15 +16,37 @@
        flex-order-gt-sm="2" flex-order-sm="1" flex-order-xs="1"
        layout="row" layout-align="start center">
     <ng-transclude id="footer-content"></ng-transclude>
-    <a class="che-footer-button-blue che-footer-button" id="footer-email-wish" ng-if="cheFooterController.supportEmail"
-       ng-href="{{'mailto:' + cheFooterController.supportEmail + cheFooterController.getWishEmailSubject(cheFooterController.productName)}}" target="_blank">Make a wish</a>
-    <a class="che-footer-button-blue che-footer-button" id="footer-email" ng-if="cheFooterController.email"
-       ng-href="{{'mailto:' + cheFooterController.email.address + cheFooterController.getEmailSubject(cheFooterController.email.subject)}}">{{cheFooterController.email.title}}</a>
-    <a class="che-footer-button-blue che-footer-button" id="footer-docs" ng-if="cheFooterController.docs" ng-href="{{cheFooterController.docs}}" target="_blank">Docs</a>
-    <a class="che-footer-button-blue che-footer-button" id="footer-support" ng-if="cheFooterController.supportHelpPath && cheFooterController.supportHelpTitle"
-       ng-href="{{cheFooterController.supportHelpPath}}" target="_blank">{{cheFooterController.supportHelpTitle}}</a>
-
-    <a ng-repeat="link in cheFooterController.links" class="che-footer-button-blue che-footer-button" ng-href="{{link.location}}" target="_blank">{{link.title}}</a>
+
+    <a ng-if="cheFooterController.links.supportEmail"
```

Comment:
The difference between `ng-if` and `ng-show` appears only when condition is `false`. Since `supportEmail` has a default value it doesn't matter which one is used.

Reply:
for example

https://github.com/eclipse/che/pull/15907/files/0415d145a2c1063b3871c2dbc188946d7b22b61c#diff-7c369f4951e48303aaf60f8b4f70026eR73

'cheBranding.getFooter().email' has a default value - 'null'
https://github.com/eclipse/che/blob/master/dashboard/src/components/branding/che-branding.factory.ts#L300
```
  getFooter(): IBrandingFooter {
    return {
      content: this.branding.footer && this.branding.footer.content ? this.branding.footer.content : '',
      links: this.branding.footer && this.branding.footer.links ? this.branding.footer.links : [],
      email: this.branding.footer && this.branding.footer.email ? this.branding.footer.email : null
    };
  }
```

Reply:
So if the default value is `null` then the node won't appear in DOM until `supportEmail` gets some truthy value. Maybe I didn't get your point but I don't see any problem here. 
It would be nice if you explain the problem that your approach solves better.

Reply:
DOM operations are very expensive.
 null will change for something else and dom will change. Am I right?

Reply:
It depends. 
In this case the condition may change only once, or may not change at all. That's why it won't cause performance problem, it's not the same as re-render something tens times per second.